packages/core/src/invoke.ts

```diff
@@ -177,15 +209,20 @@ export function spawnCallback<TE extends EventObject = AnyEventObject>(
       parent.send(error(id, err) as any);
     }
 
-    // if (isPromiseLike(callbackStop)) {
-    //   // it turned out to be an async function, can't reliably check this before calling `callback`
-    //   // because transpiled async functions are not recognizable
-    //   return this.spawnPromise(callbackStop as Promise<any>, id);
-    // }
+    if (isPromiseLike(callbackStop)) {
```

Comment:
the beauty of the approach in this PR is that this is not needed, the type of the spawned thing gets declared upfront, so a user really shouldn't pass an async function to `spawnCallback`

Reply:
Keeping that here for compatibility (tests pass) for now, will later refactor when `spawnCallback` is removed.